src/server/common.cc

```diff
@@ -318,8 +293,11 @@ OpResult<ScanOpts> ScanOpts::TryFrom(CmdArgList args) {
       if (scan_opts.pattern == "*")
         scan_opts.pattern = string_view{};
     } else if (opt == "TYPE") {
-      ToLower(&args[i + 1]);
-      scan_opts.type_filter = ArgS(args, i + 1);
+      auto obj_type_as_string = ObjTypeFromString(ArgS(args, i + 1));
+      if (!obj_type_as_string) {
+        return facade::OpStatus::SYNTAX_ERR;
+      }
+      scan_opts.type_filter = obj_type_as_string;
```

Comment:
Redis isn't replying anything with wrong types, but I'd support doing so, because we don't have modules and know all your different types ahead

Also, ironically obj_type_as_string is not a string 😄

Reply:
:skull: